packages/extension-api/src/extension-api.d.ts

```diff
@@ -1193,6 +1193,39 @@ declare module '@podman-desktop/api' {
      * button.
      */
     cancellable?: boolean;
+
+    /**
+     * You may specify a navigation object, making the task having a
+     * navigate action that the user can trigger.
+     * @example
+     * ```ts
+     * import { window, type ProgressLocation } from '@podman-desktop/api';
+     *
+     * await window.withProgress<string>(
+     *     {
+     *       location: ProgressLocation.TASK_WIDGET,
+     *       title: 'My task',
+     *       navigation: {
+     *         routeId: 'dummy-route-id',
+     *         arguments: ['hello', 'world'],
+     *       }
+     *     },
+     *     async () => {
+     *       return 'dummy result';
+     *     },
+     *   );
+     * ```
+     */
+    navigation?: {
```

Comment:
I was thinking that navigation might not be intuitive, like why should it navigate somewhere as I'm executing a task or when is it navigating.

The main goal is to provide details of the task being executed, so maybe use `details` (navigation is how it's implemented)

```ts
details?: {
   routeId: string
   routeArgs: string[]
}
```